script.js

```diff
@@ -0,0 +1,217 @@
+
+class Player {
+	constructor(name, symbol) {
+		this.name = name,
+		this.symbol = symbol	
+	}
+}
+
+const Board = ( () => {
```

Comment:
Considering my suggestion above for Player. We can simplify our board factory as such:

```javascript
const Board = () => {
  const boardArray = new Array(9).fill(null);

  const winCombinations = // same as what you already have

  const markSquare = (index, symbol) => {
    if(boardArray[index] === null){
          boardArray[index] = symbol;
    }
  };
  
  const checkWinner = (player) => {
    return winCombinations.some(combo => combo.every(element =>   player.moves.includes(element)));
  }

const checkDraw = () => {
    return boardArray.every(element => element !== null);
  }
}
```

Reply:
Thanks raynv, now it's more explicit.